src/generic/stage2/numberparsing.h

```diff
@@ -393,7 +393,7 @@ really_inline bool parse_number(UNUSED const uint8_t *const src,
   uint64_t i;      // an unsigned int avoids signed overflows (which are bad)
   if (*p == '0') { // 0 cannot be followed by an integer
     ++p;
-    if (is_not_structural_or_whitespace_or_exponent_or_decimal(*p)) {
```

Comment:
It looks like this is the only place this is called; the function should be removable now. Yay!

Reply:
I'll remove the function before merging. This is a small win, but reducing the complexity is important.